src/components/ui/CraftingRequirementsView.tsx

```diff
@@ -0,0 +1,262 @@
+import { GameState, InventoryItemName } from "features/game/types/game";
+import { ITEM_DETAILS } from "features/game/types/images";
+import React from "react";
+import { RequirementLabel } from "./RequirementsLabel";
+import { SquareIcon } from "./SquareIcon";
+
+/**
+ * The props for the component.
+ * @param wideLayout Whether to always use the wide layout for large screen or not.
+ * @param gameState The game state.
+ * @param details The item details.
+ * @param requirements The item quantity requirement.
+ * @param actionView The view for displaying the crafting action.
+ */
+interface Props {
+  wideLayout?: boolean;
+  gameState: GameState | GoblinState;
+  details: ItemDetailsProps | AchievementDetailsProps | NonItemDetailsProps;
+  requirements?: RequirementsProps;
+  actionView?: JSX.Element;
+}
+
+/**
+ * The props for the details for items.
+ * @param type The type is item.
+ * @param item The item.
+ * @param quantity The item quantity. Leave it undefined if quantity is not displayed.
+ */
+interface ItemDetailsProps {
+  type: "item";
+  item: InventoryItemName;
+  quantity?: Decimal;
+}
+
+/**
+ * The props for the details for items.
+ * @param type The type is achievement.
+ * @param achievement The achievement.
+ */
+interface AchievementDetailsProps {
+  type: "achievement";
+  achievement: BumpkinSkillName;
+}
+
+/**
+ * The props for the details for non items.
+ * @param type The type is non item.
+ * @param icon The icon.
+ * @param title The title.
+ * @param description The description.
+ */
+interface NonItemDetailsProps {
+  type: "nonItem";
+  icon?: string;
+  title?: string;
+  description?: string;
+}
+
+/**
+ * The props for the crafting requirements.
+ * @param resources The item resources requirements.
+ * @param sfl The SFL requirements.
+ * @param showSflIfFree Whether to show free SFL requirement if SFL cost is 0. Defaults to false.
+ * @param sellForSfl The amount of SFL the item can be sold for.
+ * @param level The level requirements.
+ * @param xp The XP gained for consuming the item.
+ * @param timeSeconds The wait time in seconds for crafting the item.
+ * @param harvests The min/max harvests for the item.
+ * @param showOpenSeaLink Whether to show the open sea link or not.
+ */
+interface RequirementsProps {
+  resources?: Ingredient[];
+  sfl?: Decimal;
+  showSflIfFree?: boolean;
+  sellForSfl?: Decimal;
+  level?: number;
+  xp?: Decimal;
+  timeSeconds?: number;
+  harvests?: HarvestsRequirementProps;
+  showOpenSeaLink?: boolean;
+}
+
+/**
+ * The props for harvests requirement label.
+ * @param minHarvest The minimum number of harvests.
+ * @param maxHarvest The maximum number of harvests.
+ */
+interface HarvestsRequirementProps {
+  minHarvest: number;
+  maxHarvest: number;
+}
+
+/**
+ * The view for displaying item name, details, crafting requirements and action.
+ * @props The component props.
+ */
+export const CraftingRequirementsView: React.FC<Props> = ({
```

Comment:
We are aiming for consistency here and a reusable component is great.

However this component feels like it is trying to do/be more than one thing. There are lots of if statements and conditionals scattered throughout.

I would suggest creating reusable components for specific use cases - this will ensure we can tailor and improve each component dependent on its use case without breaking other UI dependencies.

Examples of the use cases that come to mind.

- ExpansionRequirements.tsx
- CraftingRequirements
- InventoryItemDetails

Reply:
I agree with your points, seems like I am somewhat making the component too generalized.  I will work on breaking it up into a couple different components